controllers/profiles/setFollowing.js

```diff
@@ -97,11 +97,12 @@ module.exports = async (req, res) => {
             user_id: req.body.user_id_followed,
             user_id_follower: req.body.user_id_follower,
             username_follower: req.body.username_follower
-          }
+          },
+          token: userToken.token
         };
         if (userToken) {
           messaging()
-            .sendToDevice(userToken.token, payload)
+            .send(payload)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The check for `userToken` should be done before trying to access `userToken.token`. If `userToken` is null or undefined, this will throw an error. Also, the `.then(() => {})` after `messaging().send(payload)` is redundant and can be removed if no action is performed after the promise resolves.

```diff
- 101:           token: userToken.token
- 102:          };
- 103:          if (userToken) {
- 104:            messaging()
- 105:             .send(payload)
-             .then(() => {});
+ 101:          };
+ 102:          if (userToken) {
+ 103:            payload.token = userToken.token;
+ 104:            messaging()
+ 105:             .send(payload);
         }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->